dbt/adapters/bigquery/column.py

```diff
@@ -215,15 +217,29 @@ def _update_nested_column_data_types(
 
     if len(column_name_parts) == 1:
         # Base case: column is not nested - store its data_type concatenated with constraint if provided.
-        nested_column_data_types[root_column_name] = (
+        column_data_type_and_constraints = (
             column_data_type
             if column_rendered_constraint is None
             else f"{column_data_type} {column_rendered_constraint}"
         )
+        if root_column_name not in nested_column_data_types:
```

Comment:
Can we do the positive assertion first? For example:
```python
if existing_nested_column_data_type := nested_column_data_types.get(root_column_name):
    assert isinstance(existing_nested_column_data_type, dict)  # keeping mypy happy
    existing_nested_column_data_type.update({
        _PARENT_DATA_TYPE_KEY: column_data_type_and_constraints
    })
else:
    nested_column_data_types.update({
        root_column_name: column_data_type_and_constraints
    })
```

Reply:
yep -- I think I wrote it as-is because the positive assertion is more of the special case here but looking at it again after a couple days, I'm all for rearranging this.